code/modules/antagonists/heretic/magic/rust_charge.dm

```diff
@@ -0,0 +1,49 @@
+// Rust charge, added so i could put the lion hunter rifle earlier up the tree
+/datum/action/cooldown/mob_cooldown/charge/rust
+	name = "Rust Charge"
+	desc = "A charge that must be started on a rusted tile and will destroy any rusted objects you come into contact with, will deal high damage to others and rust around you during the charge."
+	charge_distance = 10
+	charge_damage = 50
+	cooldown_time = 45 SECONDS
```

Comment:
You need to either check that the heretic has a focus to cast this or indicate somewhere it does not require a focus